### PR TITLE
Add units to weather template

### PR DESCRIPTION
**Pull request by gjohansson-ST**
## Proposed change

Add units to weather template

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/74060
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Review by frenck — DISMISSED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

File: source/_integrations/weather.template.markdown [hunk 1]
```diff
@@ -54,6 +55,10 @@ temperature_template:
   description: The current temperature.
   required: true
   type: template
+temperature_unit:
+  description: Unit for temperature.
```
Comment (emontnemery) [inline source/_integrations/weather.template.markdown]: We should explain that:
- This is the unit of the template output, not what the user sees
- The output will be converted according to the user's unit system or entity override, we should explain it in https://developers.home-assistant.io/docs/core/entity/weather and link to it
- The valid options are °C, °F and K

**Review by emontnemery — CHANGES_REQUESTED** [review]

File: source/_integrations/weather.template.markdown (hunk 1 above)
Comment (gjohansson-ST) [reply source/_integrations/weather.template.markdown]: Should now be covered

File: source/_integrations/weather.template.markdown [hunk 2]
```diff
@@ -15,6 +15,7 @@ There are several powerful ways to use this integration, including localizing yo
 
 Another use case could be using temperature and humidity from one weather platform, with forecasts from a different one.
 
+Output will be converted according to the users unit system or entity override, see [documentation](https://developers.home-assistant.io/docs/core/entity/weather/#unit-conversion).
```
Comment (emontnemery) [inline source/_integrations/weather.template.markdown]:
```suggestion
Output will be converted according to the user's unit system or entity override, see [documentation](https://developers.home-assistant.io/docs/core/entity/weather/#unit-conversion).
```

**Review by emontnemery — DISMISSED** [review]
Thanks, @gjohansson-ST 👍

**Review by emontnemery — APPROVED** [review]

**Pull request merged**